sample/src/main/java/test/Hello.java

```diff
@@ -104,9 +100,37 @@ private void testAWSClientInstrumentation() {
         }
     }
 
+    private Future<Void> scanTable() {
+        // Executed in a separate thread with X-Ray context maintained
+        return executor.submit(() -> {
+            // Dynamo test
+            ScanRequest scanRequest = new ScanRequest().withTableName(DYNAMO_TABLE_NAME);
+            ScanResult ddbResult = dynamo_client.scan(scanRequest);
+            System.out.println(ddbResult);
+            return null;
+        });
+    }
+
+    public void testConcurrency() {
+        System.out.println("Starting DynamoDB Concurrency test...");
+        AWSXRay.beginSubsegment("handleDynamoRequest");
+        Future<Void> eventFuture = scanTable();
+
+        try {
+            eventFuture.get();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
why are we omitting these 2 exceptions?

Reply:
You're right, we should end the subsegment with them.